mcs/class/Mono.Data.Tds/Mono.Data.Tds.Protocol/Tds70.cs

```diff
@@ -586,6 +586,13 @@ private void WriteParameterInfo (TdsMetaParameter param)
 			} else if (colType == TdsColumnType.BigVarBinary) {
 				if (size > 8000)
 					colType = TdsColumnType.Image;
+			} else if (colType == TdsColumnType.DateTime2 ||
+				   colType == TdsColumnType.DateTimeOffset) {
+				// HACK: Wire-level DateTime{2,Offset}
+				// require TDS 7.3, which this driver
+				// does not implement correctly--so we
+				// serialize to ASCII instead.
```

Comment:
Hi Atsushi,

I haven't investigated fully, but have written my conclusions in the
commit message: https://github.com/ztzg/mono/commit/f055c46a